src/hotspot/share/utilities/decoder.hpp

```diff
@@ -122,6 +122,9 @@ class Decoder : AllStatic {
 
   static void print_state_on(outputStream* st);
 
+#ifdef _WINDOWS
```

Comment:
I'm not sure we have to use #ifdef WINDOWS here. 
There is src\hotspot\os\windows\decoder_windows.cpp file already, I beileve it should be there.

Reply:
The thing is that `decoder.cpp` has most of the code guarded by similar ifdefs. I don't think it's right to expose something in headers that would fail during linking, so I've put most of it to the else branch.
Yes, I could have used just a declaration in here and added the noop impl to `decoder_windows.cpp`; this way I am modifying just 1 file rather than 2.